docs/proposals/params.md

```diff
@@ -0,0 +1,205 @@
+    self-bootstrap with just a master URL
+2.  Users could model external shared services and consume information about them in pods without
+    having to redefine parameters in each container
+3.  Users could model supplemental information about services as parameters and consume them in the
+    pods that consumed those services
+4.  Users could store configuration info for their components as parameters and consume them in the
+    appropriate pods
+
+## Use-Case: Configuring system components
+
+Upstream issue [kubernetes/1627](https://github.com/GoogleCloudPlatform/kubernetes/issues/1627)
+discusses storing configuration for system components in a resource with a `map[string]string`
+structure.  If a component is configured with a namespace/name, it should be able to pull its
+configuration from the API server.  A `Parameters` resource would solve this use case; a system
+component would only need the URL for the API server and the coordinates for its configuration
+in order to discover its configuration.  It could also watch the `Parameters` resource and receive
+events when its configuration was changed.
+
+An alternative to a parameters resource that has been discussed for configuring system components
+upstream is to store configuration data in etcd directly.  Using `Parameters` to sure
+configuration about services has several advantages to storing them directly in etcd or using CLI
+arguments exclusively:
+
+1.  Eliminates the need to query etcd directly, resulting in lower complexity; consumers only need
+    to know how to query the API server instead of the API server and etcd
+2.  Keeps invocation simple by eliminating the need to pass tons of CLI arguments; components only
+    need to be configured with the coordinates of the API server and their namespace/name
+3.  Allows components to easily watch for configuration changes and adapt to new configurations
+    without restart
+
+## Use-Case: Metadata about services
+
+A `Parameters` resource can also hold information about how to use a service.  For example, a
+service that provides a mysql database could store the database name and username as parameters.
+Pods consuming the mysql service would also consume the mysql parameters instance in order to be
+configured with the username and database name.  Declaring pod dependencies on services has been
+proposed upstream in [kubernetes/1768](https://github.com/GoogleCloudPlatform/kubernetes/issues/1768)
+via `ServiceLink` type.  A similar notion could be introduced for `Parameters` -- containers in a
+pod would declare `ParametersLink`s to indicate that a container consumed the referenced
+`Parameters` instance.
+
+## Use-Case: External services
+
+A `Parameters` resource can also be used to model information about an external service which is
+shared amongst pods in a namespace.  For example, if many pods in a namespace need information
```

Comment:
Username/password is clearly secrets.  Is there a different type of value for an external service that is not the same as database/username/password?